pinot-query-planner/src/main/java/org/apache/pinot/query/routing/ServerAddress.java

```diff
@@ -26,10 +26,16 @@ public class ServerAddress {
 
```

Comment:
Can we add a javadoc to describe the exact role of `ServerAddress`?

Also can you clarify why we need both `VirtualServer` and `ServerAddress`?

Reply:
`ServerAddress` is just hostname:port:partition (or whatever I rename it to)

`VirtualServer` is essentially a `ServerAddress` + `ServerInstance`, which has a lot more information than just the hostname/port/partition. I didn't model it like that because the information in `ServerAddress` would be redundant. I couldn't just use `VirtualServer` throughout the code base because there are some places where we don't have all the information we need to build an instance of `ServerInstance` (and therefore `VirtualServer`)